modular_skyrat/modules/modular_implants/code/nifs.dm

```diff
@@ -0,0 +1,511 @@
+	var/broken = FALSE
+	///Does the NIF have theft protection? This should only be disabled if admins need to fix something.
+	var/theft_protection = TRUE
+	///Is the NIF able to take damage?
+	var/durability_loss_vulnerable = TRUE
+
+	//Software Variables
+	///How many programs can the NIF store at once?
+	var/max_nifsofts = 5
+	///What programs are currently loaded onto the NIF?
+	var/list/loaded_nifsofts = list()
+	///What programs come already installed on the NIF?
+	var/list/preinstalled_nifsofts = list()
+	///This shows up in the NIF settings screen as a way to ICly display lore.
+	var/manufacturer_notes = "There is no data currently avalible for this product"
+
+	//Appearance Variables
+	///This is the sound that plays when doing most things!
+	var/good_sound ='modular_skyrat/modules/modular_implants/sounds/default_good.ogg'
+	///This is the sound that plays if there is an issue going on.
+	var/bad_sound = 'modular_skyrat/modules/modular_implants/sounds/default_bad.ogg'
+	///This is the sound that you would hear if you enable if you activate or enable something.
+	var/click_sound = 'modular_skyrat/modules/modular_implants/sounds/default_click.ogg'
+	///What icon does the NIF display in chat when sending out alerts? Icon states are stored in 'modular_skyrat/modules/modular_implants/icons/chat.dmi'
+	var/chat_icon = "standard"
+
+/obj/item/organ/internal/cyberimp/brain/nif/Initialize(mapload)
+	. = ..()
+
+	durability = max_durability
+	loaded_nifsofts = preinstalled_nifsofts
+
+	for(var/datum/nifsoft/preinstalled_nifsoft as anything in preinstalled_nifsofts)
+		install_nifsoft(preinstalled_nifsoft)
+
+	power_level = max_power
+
+/obj/item/organ/internal/cyberimp/brain/nif/Insert(mob/living/carbon/human/insertee)
+	. = ..()
+
+	if(linked_mob && stored_ckey != insertee.ckey && theft_protection)
+		insertee.audible_message(span_warning("The [src] lets out a negative buzz before forcefully removing itself from [insertee]'s brain."))
+		playsound(insertee, 'sound/machines/buzz-sigh.ogg', 30, TRUE)
+		src.Remove(insertee)
+		src.forceMove(get_turf(insertee))
+
+		return FALSE
+
+	linked_mob = insertee
+	stored_ckey = linked_mob.ckey
+
+	loc = insertee // This needs to be done, otherwise TGUI will not pull up.
+	insertee.installed_nif = src
+	START_PROCESSING(SSobj, src)
+
+	if(!is_calibrated)
+		send_message("Calibration Process Starting!")
+		calibrating = TRUE
+
+/obj/item/organ/internal/cyberimp/brain/nif/Remove(mob/living/carbon/organ_owner, special = FALSE)
+	. = ..()
+
+	organ_owner.log_message("[src] was removed from [organ_owner]]",LOG_GAME)
+	STOP_PROCESSING(SSobj, src)
+
+/obj/item/organ/internal/cyberimp/brain/nif/process(delta_time)
+	. = ..()
+
+	if(!linked_mob || broken)
+		return FALSE
+
+	if(calibrating)
+		preform_calibration()
+
+	if(IS_IN_STASIS(linked_mob))
+		return
+
+	if((linked_mob.nutrition < minimum_nutrition) && (nutrition_drain)) //Turns nutrition drain off if nutrition is lower than minimum
```

Comment:
Also, you'll need to account for mobs that don't use nutrition, too, since those exist.
```suggestion
	if(nutrition_drain && linked_mob.nutrition < minimum_nutrition) //Turns nutrition drain off if nutrition is lower than minimum
```

Reply:
The settings menu prevents a mob from using this option if they lack nutrition.